### PR TITLE
Wait elasticsearch job done to call `flush` callback.

### PR DESCRIPTION
**Pull request by tadjik1**
In some usecases it's very important to know when exactly elasticsearch stream finishes. In order to achieve this we could receive callback to `flush function` and call it `after` elasticsearch job will finish. 
#49

### CODE REVIEW

**Comment by orangejulius** [comment]
Hey @tadjik1,
This is great! I could see us using this in the Pelias code, so I'd like to get this merged. Would it be possible for you to create a short but fully working code snippet that shows how this would be used? Feel free to either just do it as a comment here or as a change in the README file, which is where I'd like it to end up eventually.

**Comment by tadjik1** [comment]
@orangejulius sure! will update PR soon

**Comment by tadjik1** [comment]
@orangejulius 
Please check my example in readme. I added there a real case, when we wanted to remove outdated documents after import was done

File: src/BatchManager.js [hunk 1]
```diff
@@ -94,35 +94,42 @@ BatchManager.prototype._dispatch = function( batch ){
     this._attemptResume();
     this._attemptEnd();
 
+    if (next) {
```
Comment (missinglink) [inline src/BatchManager.js]: can we make this `'function' === typeof next`?

File: src/BatchManager.js [hunk 2]
```diff
@@ -94,35 +94,42 @@ BatchManager.prototype._dispatch = function( batch ){
...
   }
 };
 
-BatchManager.prototype._attemptEnd = function(){
+BatchManager.prototype._attemptEnd = function(next){
   if( this.finished && !this._transient && !this._current._slots.length ){
     this._opts.client.close();
     stats.end();
+
+    if (next) {
```
Comment (missinglink) [inline src/BatchManager.js]: here too

**Comment by tadjik1** [comment]
@missinglink done

**Pull request merged**